### ISSUE TITLE
build broken: gRPC versions don't match

### ISSUE THREAD

**Issue by cuongdo** [cockroachdb/examples-go#73]
Example broken build:

https://circleci.com/gh/cockroachdb/examples-go/196

Relevant log excerpt:

```
# github.com/cockroachdb/cockroach/roachpb
../.go_workspace/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:1486: undefined: grpc.SupportPackageIsVersion2
```

It looks like the `cockroach` repo depends on gRPC v2 (via its `GLOCKFILE`). Since `examples-go` has no `GLOCKFILE`, it's pulling the latest version from master, which is on v3:

https://github.com/grpc/grpc-go/blob/master/rpc_util.go#L449

We probably should upgrade our gRPC, but this problem probably would have been avoided/deferred with some kind of vendoring.

**Comment by cuongdo** [comment]
@tamird: what are the risks in upgrading the version of `grpc-go` we use for `cockroach`?

**Comment by petermattis** [comment]
@cuongdo I believe we're using gRPC APIs that have been removed. So we can't just upgrade `grpc-go` for `cockroach` without replacing our usage of those APIs (which is non-trivial).

**Comment by tamird** [comment]
Correct. There's an issue tracking this somewhere. I believe @d4l3k is
assigned.

**Comment by d4l3k** [comment]
For reference, https://github.com/cockroachdb/cockroach/issues/7494 https://github.com/cockroachdb/cockroach/issues/7430
